src/ai/configuration.cpp

```diff
@@ -111,16 +111,16 @@ void extract_ai_configurations(std::map<std::string, description> &storage, cons
 }
 }
 
-void configuration::add_era_ai_from_config(const config &era)
+void configuration::add_era_ai_from_config(const config& era)
 {
 	era_ai_configurations_.clear();
 	extract_ai_configurations(era_ai_configurations_, era);
 }
 
-void configuration::add_mod_ai_from_config(config::const_child_itors mods)
+void configuration::add_mod_ai_from_config(const config::const_child_itors& mods)
```

Comment:
I think (not fully sure) that config::const_child_itors is a type with the trivial destructor that should usually be passed by value. Can you give more information why it suggests this change?

Reply:
Maybe a question for @jj-marr as well?

Reply:
`static_assert(std::is_trivially_copyable_v<config::const_child_itors>);` passes on gcc. Does it pass on clang?

Reply:
More specifically, the check itself says:
> The check is only applied to parameters of types that are expensive to copy which means they are not trivially copyable or have a non-trivial copy constructor or destructor.
A trivially copyable type must have a trivial copy constructor and destructor, I believe.
Both clang and gcc seem to identify `boost::iterator_range` correctly as trivially copyable:
https://godbolt.org/z/s8c9oxzGn
Perhaps the better solution is to explicitly whitelist `boost::iterator_range`? There's an `AllowedTypes` option for this check.
https://clang.llvm.org/extra/clang-tidy/checks/performance/unnecessary-value-param.html

Reply:
> I think (not fully sure) that config::const_child_itors is a type with the trivial destructor that should usually be passed by value. Can you give more information why it suggests this change?

> Perhaps the better solution is to explicitly whitelist boost::iterator_range? There's an AllowedTypes option for this check.

If any other changes are needed, it's best done by @jj-marr in a follow-up PR. I was simply trying to get the broken CI back into working order.